src/main/java/com/aws/greengrass/shadowmanager/sync/model/CloudUpdateSyncRequest.java

```diff
@@ -86,86 +86,88 @@ public void merge(CloudUpdateSyncRequest other) {
     public void execute(SyncContext context) throws RetryableException, SkipSyncRequestException,
             ConflictException, UnknownShadowException, InterruptedException {
 
-        Optional<ShadowDocument> shadowDocument = context.getDao().getShadowThing(getThingName(), getShadowName());
-
-        //TODO: store this information in a return object to avoid unnecessary calls to DAO.
-        SyncInformation currentSyncInformation = context.getDao()
-                .getShadowSyncInformation(getThingName(), getShadowName())
-                .orElseThrow(() -> new UnknownShadowException("Shadow not found in sync table"));
+        synchronized (context.getSynchronizeHelper().getThingShadowLock(this)) {
```

Comment:
do you also need to synchronize in `isUpdateNecessary(SyncContext context)` ?

Reply:
technically no but it certainly wouldn't hurt.  that method is only ever called within this synchronize block https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/main/src/main/java/com/aws/greengrass/shadowmanager/ipc/UpdateThingShadowRequestHandler.java#L111.

Reply:
just saw the edit about adding a comment, will do

Reply:
updated:  when answering the question of "why not needed for other types of requests" i think the same issue would happen for LocalUpdateSyncRequest, since `isUpdateNecessary` which reads/writes to sync table, is called in context of mqtt callback thread and the sync thread

Reply:
however if we need to have similar synchronization for LocalUpdateSyncRequest, there's the issue where we can't block the subscribe callback thread in https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/8e24c7de2f5d091fb6a42247742d28be76f96a5a/src/main/java/com/aws/greengrass/shadowmanager/sync/CloudDataClient.java#L279

Reply:
i guess we could dispatch an async task, but now things are becoming more complex